src/abstractprobprog.jl

```diff
@@ -60,3 +61,23 @@ m = decondition(condition(m, obs))
 should hold for generative models `m` and arbitrary `obs`.
 """
 function condition end
+
+
+"""
+    rand([rng=Random.default_rng()], [T=NamedTuple], model::AbstractProbabilisticProgram) -> T
+
+Draw a sample from the prior and prior-predictive distribution of the model specified by the
+probabilistic program.
+
+The sample will be returned as format specified by `T`.
+"""
+Base.rand(rng::Random.AbstractRNG, ::Type, model::AbstractProbabilisticProgram)
```

Comment:
I don't understand why this is the recommended approach, since it defines a fallback that returns a `nothing` instead of raising a `MethodError`:
```julia
julia> using AbstractPPL, Random

julia> struct Model <: AbstractProbabilisticProgram end;

julia> rand(Random.default_rng(), NamedTuple, Model())

julia> @which rand(Random.default_rng(), NamedTuple, Model())
rand(rng::AbstractRNG, ::Type, model::AbstractProbabilisticProgram)
     @ AbstractPPL ~/projects/AbstractPPL.jl/src/abstractprobprog.jl:74

julia> @code_lowered rand(Random.default_rng(), NamedTuple, Model())
CodeInfo(
1 ─     return nothing
)
```

Reply:
It's mentioned here: https://docs.julialang.org/en/v1/manual/documentation/#Functions-and-Methods-2

The example there does not define a fallback method but only defines a function with 0 methods (and I would say that's preferable to an undefined variable):
```julia
julia> """
       A docstring
       """
       function f end
f

julia> f
f (generic function with 0 methods)

julia> """
       A docstring
       """
       g
g

julia> g
ERROR: UndefVarError: g not defined
```
However, apparently when you add arguments that a fallback method is defined:
```julia
julia> """
       A docstring
       """
       function h(x) end
h

julia> h
h (generic function with 1 method)
```

Reply:
Since you have a concrete method with a concrete list of arguments for which you want to add a docstring here, maybe the most reasonable approach would be to just add the docstring to the definition of `rand(::AbstractRNG, ::Type{T}, ::AbstractProbabilisticProgram)`?

Reply:
I thought that's what the original version was doing. How would it look different?

Reply:
No, it would not repeat the signature but only add the docstring to the actual implementation. I'll make a suggestion below.

Reply:
Ah, I only noticed now that there is no fallback definition here. Somehow I assumed that you implement a function with this signature below, and then it would have seemed better to add the docstring there.